### ISSUE TITLE
Add functionality to change locale in VS Code test instance

### ISSUE THREAD

**Issue by SounD120** [Microsoft/vscode-extension-vscode#136]
Hello. I am from [vscode-react-native](https://github.com/Microsoft/vscode-react-native). Recently we have implemented extension localization via [vscode-nls](https://github.com/Microsoft/vscode-nls) and [vscode-nls-dev](https://github.com/Microsoft/vscode-nls-dev) npm packages, and now we want to cover it with tests. Standard `test` script doesn't contain functionality to change locale in the test VS Code instance, but [according to documentation](https://code.visualstudio.com/docs/editor/command-line#_core-cli-options) VS Code supports `--locale` startup key to run with the specific locale. We tried to apply hack by using `CODE_TEST_WORKSPACE` as locale argument by this Bash command in our Azure DevOps Windows 10 CI
```
export CODE_TEST_WORKSPACE=--locale=ru
```
so the tests startup command looks like
```
d:\_work\1\s\.vscode-test\stable\Code.exe --locale=ru --extensionDevelopmentPath=d:\_work\1\s --extensionTestsPath=D:/_work/1/s/test/extension/localization
```
But `--locale` key is not working if we apply it that way and our tests fails due to language is not being switched.
So we'd like to clarify is this possible to change VS Code locale via test script arguments in CI, or at least if it possible to add this argument to the `vscode` package test script?

**Comment by bpasero** [comment]
Adding @dbaeumer, I think the `--locale` switch should work even in extension tests if I am not mistaken.

**Comment by dbaeumer** [comment]
The `--locale` switch will work for VS Code and if a Russian language pack is installed it should respect this. It might be different for your extension. The JS code needs to be compiled for production (the JS rewriting needs to have taken place). Otherwise the JS code you are executing during test will not respect the locale setting.

**Comment by dbaeumer** [comment]
The JS rewriting happens via the gulp tasks provided from `vscode-nls-dev`. I looked at your reprository and the `build` gulp tasks seems to do this. So all necessary translation files will be created.

When you start VS Code in the tests with a --locale param and an installed language pack for that locale will it come up in that locale?

**Comment by SounD120** [comment]
@bpasero , @dbaeumer As we understand the VS Code localization mechanism, it doesn't matter if we have installed language pack for the extensions localized with `vscode-nls` or not. Localization works if we run `code --locale=ru` locally, but if we try to apply `--locale` parameter in CI it is not, so we actually don't know how to switch VS Code language with [test script](https://github.com/Microsoft/vscode-extension-vscode/blob/master/bin/test). Also, test script doesn't provide extra parameter for running it with another locale and **we had to switch it with our** [**patched version**](https://github.com/SounD120/vscode-react-native/blob/localization-tests/tools/test1#L51-L69) **during CI builds**. With respect to JS rewriting, at the moment of testing all the localization bundles are already built, and the `localize` calls are rewrited, so it should not affect the language switching.

**Comment by bpasero** [comment]
@SounD120 how about a PR to fix the issue for everyone if it works for you?

**Comment by SounD120** [comment]
@bpasero The problem is that it is **not** work for us.

**Comment by bpasero** [comment]
Oh ok, your wording made it sound as if it was working.

**Comment by ruslan-bikkinin** [comment]
@bpasero summarizing all things:
* test script doesn't provide an option to run extension tests under the VS Code instance with another locale
* [possible fix](https://github.com/SounD120/vscode-react-native/blob/d1bb7f9ff4edcd548a539a2c0d9c7c965a3ec020/tools/test1#L64) for it doesn't work for us for some reason

So we are asking for help to get it work for us.

**Comment by bpasero** [comment]
@ruslan-bikkinin I still think the change to the `test` file would be valuable so that the `--locale` is passed to the instance. For example with the following steps I am able to bring up a test instance in german locale (for VSCode at least):
* install the german language pack in the instance of VSCode that the test script is downloading
* add `--locale=de` to the arguments in the `test` file
* run `npm run test` from the command line

Now, I am not the right person to help you with how translations are installed for extensions. Waiting for @dbaeumer to comment on what you need to do to install localizations for an extension.

**Comment by dbaeumer** [comment]
In @SounD120 case the localization are bundled with the extension since the are not part of the core VS drop and therefore not part of the language packs. So IMO if the test instance starts in German locale and the extension is compiled for productions (produces localization files) and has a German translation the strings should show up in German.

**Comment by bpasero** [comment]
@SounD120 is that true?

**Comment by SounD120** [comment]
@bpasero , @dbaeumer I sent you a direct email with the details.

**Comment by SounD120** [comment]
@bpasero , @dbaeumer Thanks a lot for your help, guys. We found the way to solve our issue, so I proposed a PR that adds locale switching functionality to `test` script

**Comment by SounD120** [comment]
@bpasero Could you please clarify when the new version of the package will be released?

**Comment by bpasero** [comment]
@SounD120 I just released `1.1.27`
